单元测试失败: TaskStateManager.listTaskHistory 历史记录顺序错误

## 测试输出错误信息

```
 FAIL  src/utils/task-state-manager.test.ts > TaskStateManager > listTaskHistory > should list completed tasks
AssertionError: expected 'Task 1' to be 'Task 2' // Object.is equality

Expected: "Task 2"
Received: "Task 1"

 ❯ src/utils/task-state-manager.test.ts:243:33
    241|       const history = await listManager.listTaskHistory();
    242|       expect(history.length).toBe(2);
    243|       expect(history[0].prompt).toBe('Task 2'); // Most recent first
       |                                 ^
    244|       expect(history[1].prompt).toBe('Task 1');
```

## 根本原因分析

测试 `listTaskHistory > should list completed tasks` 验证任务历史记录应按时间倒序返回（最新的在前）：
- 先完成 Task 1，再完成 Task 2
- 期望 `history[0].prompt` 为 `'Task 2'`（最新）
- 实际返回 `'Task 1'`（最旧）

**可能原因:**
1. `listTaskHistory()` 方法返回的数组顺序与预期不符，可能是按时间正序而非倒序
2. 新提交 `3ba4ea9` (feat(agent): dynamic Agent Prompt adaptation based on channel capabilities) 可能影响了某些底层逻辑

## 触发提交

`3ba4ea9` feat(agent): dynamic Agent Prompt adaptation based on channel capabilities (Issue #590 Phase 3) (#630)

## 相关文件

- 测试文件: `src/utils/task-state-manager.test.ts`
- 源文件: 可能是 `src/utils/task-state-manager.ts`

## CI 运行时间

2026-03-05 09:30:56 +0800

--- a/src/utils/task-state-manager.ts
+++ b/src/utils/task-state-manager.ts
@@ -360,9 +360,18 @@ export class TaskStateManager {
         }
       }
 
-      // Sort by creation date (newest first) and limit
+      // Sort by completion time (newest first) and limit
+      // Use updatedAt since it reflects when the task was completed/cancelled
+      // Secondary sort by createdAt for stable ordering when updatedAt is equal
       return tasks
-        .sort((a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime())
+        .sort((a, b) => {
+          const updatedAtDiff = new Date(b.updatedAt).getTime() - new Date(a.updatedAt).getTime();
+          if (updatedAtDiff !== 0) {
+            return updatedAtDiff;
+          }
+          // When updatedAt is equal, sort by createdAt (newest first)
+          return new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime();
+        })
         .slice(0, limit);
     } catch {
       return [];

